routing/routing_tests/applying_traffic_test.cpp

```diff
@@ -0,0 +1,215 @@
+
+//                        Finish
+//  3        *              *
+//             ↖          ↗
+//              F5     F6
+//                ↖   ↗
+// 2 *              *
+//    ↖          ↗   ↖
+//      F2      F3      F4
+//        ↖  ↗           ↖
+// 1        *               *
+//        ↗  ↖             ^
+//      F0      F1          F8
+//    ↗          ↖         |
+// 0 *              *--F7--->*
+//   0       1      2       3
+//                Start
+// Note. This graph contains of 9 one segment directed features.
+unique_ptr<IndexGraph> BuildXXGraph(shared_ptr<EdgeEstimator> estimator)
+{
+  unique_ptr<TestGeometryLoader> loader = make_unique<TestGeometryLoader>();
+  loader->AddRoad(0 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{0.0, 0.0}, {1.0, 1.0}}));
+  loader->AddRoad(1 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{2.0, 0.0}, {1.0, 1.0}}));
+  loader->AddRoad(2 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{1.0, 1.0}, {0.0, 2.0}}));
+  loader->AddRoad(3 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{1.0, 1.0}, {2.0, 2.0}}));
+  loader->AddRoad(4 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{3.0, 1.0}, {2.0, 2.0}}));
+  loader->AddRoad(5 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{2.0, 2.0}, {1.0, 3.0}}));
+  loader->AddRoad(6 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{2.0, 2.0}, {3.0, 3.0}}));
+  loader->AddRoad(7 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{2.0, 0.0}, {3.0, 0.0}}));
+  loader->AddRoad(8 /* featureId */, true /* oneWay */, 1.0 /* speed */,
+                  RoadGeometry::Points({{3.0, 0.0}, {3.0, 1.0}}));
+
+  vector<Joint> const joints = {
+      MakeJoint({{0 /* feature id */, 0 /* point id */}}), /* joint at point (0, 0) */
+      MakeJoint({{1, 0}, {7, 0}}),                         /* joint at point (2, 0) */
+      MakeJoint({{0, 1}, {1, 1}, {2, 0}, {3, 0}}),         /* joint at point (1, 1) */
+      MakeJoint({{2, 1}}),                                 /* joint at point (0, 2) */
+      MakeJoint({{3, 1}, {4, 1}, {5, 0}, {6, 0}}),         /* joint at point (2, 2) */
+      MakeJoint({{4, 0}, {8, 1}}),                         /* joint at point (3, 1) */
+      MakeJoint({{5, 1}}),                                 /* joint at point (1, 3) */
+      MakeJoint({{6, 1}}),                                 /* joint at point (3, 3) */
+      MakeJoint({{7, 1}, {8, 0}}),                         /* joint at point (3, 0) */
+  };
+
+  unique_ptr<IndexGraph> graph = make_unique<IndexGraph>(move(loader), estimator);
+  graph->Import(joints);
+  return graph;
+}
+
+class TrafficInfoGetterTest : public TrafficInfoGetter
```

Comment:
I guess TrafficInfoGetterForTesting is even better. But I asked @bykoianko to get rid of two individual implementations and use real TrafficInfoGetter instead.